src/renderers/shared/fiber/__tests__/ReactIncremental-test.js

```diff
@@ -183,6 +183,49 @@ describe('ReactIncremental', () => {
     expect(ops).toEqual(['Foo', 'Bar', 'Bar']);
   });
 
+  it('should call callbacks even if updates are aborted', () => {
+    const callbackList = [];
+    let inst;
+
+    class Foo extends React.Component {
+      constructor(props) {
+        super(props);
+        this.state = {
+          text: 'foo',
+          text2: 'foo',
+        };
+        inst = this;
+      }
+      render() {
+        return (
+          <div>
+            <div>{this.state.text}</div>
+            <div>{this.state.text2}</div>
+          </div>
+        );
+      }
+    }
+
+    ReactNoop.render(<Foo />);
+    ReactNoop.flush();
+
+    inst.setState({text: 'bar'}, () => callbackList.push('text'));
+
+    // Flush part of the work
+    ReactNoop.flushDeferredPri(20 + 5);
```

Comment:
Can you add an assertion to make sure the update processed, but wasn't committed yet? Maybe by passing an updater function to setState instead of a plain object.